app/models/game.rb

```diff
@@ -1,3 +1,34 @@
 class Game < ActiveRecord::Base
   scope :available, -> { where('black_player IS NULL OR white_player IS NULL') }
+  has_many :users
+  has_many :pieces
+
+  def populate_board!
```

Comment:
I think it would be easier to understand as a separate helper method that is called, but the way you used 2D arrays for this is very cool.